### PR TITLE
test-repart: find sfdisk

### LINKED ISSUE #21972: test-repart                               FAIL
Version: 250

Building the systemd package on Debian with repart support enabled, I get the following failure while running the test suite:

```
329/620 test-repart                                        FAIL              0.05s   exit status 1
>>> MALLOC_PERTURB_=170 /home/michael/debian/build-area/repart/systemd-249.7/src/partition/test-repart.sh /home/michael/debian/build-area/repart/systemd-249.7/build-deb/systemd-repart
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
stdout:
### Testing systemd-repart --empty=create ###
stderr:
+ set -o pipefail
+ [[ -e /dev/loop-control ]]
+ repart=/home/michael/debian/build-area/repart/systemd-249.7/build-deb/systemd-repart
+ test -x /home/michael/debian/build-area/repart/systemd-249.7/build-deb/systemd-repart
++ mktemp --tmpdir --directory test-repart.XXXXXXXXXX
+ D=/tmp/test-repart.IDXz1hFWgS
+ trap 'rm -rf '\''/tmp/test-repart.IDXz1hFWgS'\''' EXIT INT QUIT PIPE
+ mkdir -p /tmp/test-repart.IDXz1hFWgS/definitions
+ SEED=e2a40bf9-73f1-4278-9160-49c031e7aef8
+ echo '### Testing systemd-repart --empty=create ###'
+ /home/michael/debian/build-area/repart/systemd-249.7/build-deb/systemd-repart /tmp/test-repart.IDXz1hFWgS/zzz --empty=create --size=1G --seed=e2a40bf9-73f1-4278-9160-49c031e7aef8
Sized '/tmp/test-repart.IDXz1hFWgS/zzz' to 1.0G.
Applying changes.
Wiped block device.
Discarded entire block device.
Writing new partition table.
All done.
+ sfdisk -d /tmp/test-repart.IDXz1hFWgS/zzz
/home/michael/debian/build-area/repart/systemd-249.7/src/partition/test-repart.sh: line 23: sfdisk: command not found
+ grep -v -e sector-size -e '^$'
+ rm -rf /tmp/test-repart.IDXz1hFWgS
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――


...

Summary of Failures:

329/620 test-repart                               FAIL              0.05s   exit status 1


Ok:                 603 
Expected Fail:      0   
Fail:               1   
Unexpected Pass:    0   
Skipped:            16  
Timeout:            0   
```

sfdisk is installed as `/sbin/sfdisk` on Debian derived distros, which means it's not in PATH for regular users.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #21972.

cc @mbiebl.

### CODE REVIEW

File: src/partition/test-repart.sh [hunk 1]
```diff
@@ -8,6 +8,15 @@ set -o pipefail
 repart="${1:?}"
 test -x "$repart"
 
+sfdisk=
+for i in /usr/bin /usr/sbin /bin /sbin; do
```
Comment (bluca) [inline src/partition/test-repart.sh]: maybe just append /usr/sbin to PATH?

File: src/partition/test-repart.sh [hunk 2]
```diff
@@ -214,5 +214,5 @@ fi
 
 echo "### Testing json output ###"
 "$repart" "$D/zzz" --size=3G --dry-run=no --seed="$SEED" --definitions="$D/definitions" --json=help
-"$repart" "$D/zzz" --size=3G --dry-run=no --seed="$SEED" --definitions="$D/definitions" --json=pretty
+"$repart" "$D/zzz" --size=3G --dry-run=no --seed="$SEED" --definitions="$D/definitions" --json=pretty --no-pager
 "$repart" "$D/zzz" --size=3G --dry-run=no --seed="$SEED" --definitions="$D/definitions" --json=short
```
Comment (keszybz) [inline src/partition/test-repart.sh]: Shouldn't this be done in all cases?

File: src/partition/test-repart.sh (hunk 1 above)
Comment (keszybz) [reply src/partition/test-repart.sh]: Yeah, maybe just do
`command -v sfdisk >/dev/null || PATH=$PATH:/sbin:/usr/sbin` ?

Comment (bluca) [reply src/partition/test-repart.sh]: I'd just do it unconditionally, i don't think there's any purpose in not having sbin in the path these days

**Comment by yuwata** [comment]
@bluca and @keszybz Thank you for the comments. The suggestions are applied. PTAL.

**Review by bluca — APPROVED** [review]

**Comment by mbiebl** [comment]
thanks @yuwata

**Pull request merged**